samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -144,6 +129,78 @@ class DailyCallManager {
     }
   }
 
+  /**
+   * Handles participant-joined, participant-updated, and participant-left events:
+   * - Updates the participant count
+   * - Adds or removes video and audio tracks for the participant based on the event and track state
+   * @param {Object} event - The participant-* event object.
+   */
+  handleNewParticipantState(event) {
+    // Update the participant count
+    this.updateAndDisplayParticipantCount();
+
+    // Get the audio and video tracks and participant ID from the participant-* events
+    const tracks = event.participant.tracks;
+    const participant = event.participant;
+    const participantId = participant.session_id;
+
+    if (event.action === 'participant-left') {
+      if (participant) {
+        // Remove the video container if it exists
+        const videoContainer = document.getElementById(
+          `video-container-${participantId}`
+        );
+        if (videoContainer) {
+          this.destroyTrack('video', participantId);
+          videoContainer.remove();
+        }
+
+        // Remove the audio element if it exists
+        const audioElement = document.getElementById(`audio-${participantId}`);
+        if (audioElement) {
+          this.destroyTrack('audio', participantId);
+        }
+      }
+    } else {
+      for (const trackType in tracks) {
+        if (tracks[trackType].state === 'playable') {
+          if (trackType === 'video') {
+            this.displayVideo(tracks[trackType], participantId);
+          } else if (trackType === 'audio') {
+            this.playAudio(tracks[trackType], participant);
+          } else {
+            // Actions for screenVideo and screenAudio tracks if needed.
+          }
+        } else if (
+          (tracks[trackType].state === 'off') |
+          'interrupted' |
+          'blocked'
+        ) {
+          if (trackType === 'audio' && participant.local) {
+            // Do nothing for local client's audio track.
+            return;
+          } else {
+            this.destroyTrack(trackType, participantId);
+          }
+        } else {
+          // Other states are 'loading' and 'sendable'.
+          // This is a placeholder for handling those states if needed.
```

Comment:
there's a subtle thing going on here. by doing nothing for these states, this code is essentially saying "keep the whatever previous UI state we were in, whatever it was". so we could be showing _different_ UI for "sendable", for example, depending on whether the track was "off" or "playable" before. i don't think that's quite what we want.

how i would recommend structuring the `handleNewParticipantState` logic:
* first, decide whether we have a track to attach to the DOM element (if we have a `persistentTrack`, go ahead and attach it and play it)
* then, based on `state`, decide whether we want to actually display the element or show some "off" state overlay (or, in this simple example, simply hide the element)

that would solve another issue in your code here, which is that you `destroyTrack()` in the "interrupted" state, which we should avoid if there's a `persistentTrack` available—in the audio case in particular, we want to keep the audio track attached to the DOM to avoid a Safari issue where if the app is in a background tab and a remote participant leaves the "interrupted" state, they won't be audible until the tab is foregrounded.